### PR TITLE
Further indent multiline lambda's

### PR DESCRIPTION
**Pull request by nojaf**
Improves https://github.com/fsprojects/fantomas/issues/523
Not sure if I covered all cases with these changes?

### CODE REVIEW

File: src/Fantomas.Tests/CommentTests.fs [hunk 1]
```diff
@@ -43,8 +43,8 @@ let print_30_permut() =
     /// declare and initialize
     let permutation: int array =
         Array.init n (fun i ->
-            Console.Write(i + 1)
-            i)
+                Console.Write(i + 1)
```
Comment (jindraivanek) [inline src/Fantomas.Tests/CommentTests.fs]: IMO we don't want double indentation here

File: src/Fantomas/Context.fs [hunk 2]
```diff
@@ -405,17 +405,35 @@ let internal futureNlnCheckMem = Cache.memoizeBy (fun (f, ctx : Context) -> Cach
...
-    let withoutStringConst = 
+    let withoutStringConst =
         str.Replace("\\\\", System.String.Empty).Replace("\\\"", System.String.Empty).Split([|'"'|])
         |> Seq.indexed |> Seq.filter (fun (i, _) -> i % 2 = 0) |> Seq.map snd |> String.concat System.String.Empty
-    let lines = withoutStringConst.Split([|Environment.NewLine|], StringSplitOptions.None) 
+    let lines = withoutStringConst.Split([|Environment.NewLine|], StringSplitOptions.None)
+
+    (lines |> Seq.length) >= 2 || writer.Column > ctx.Config.PageWidth
+
+let internal futureNlnCheckNonCached f ctx =
```
Comment (jindraivanek) [inline src/Fantomas/Context.fs]: DRY: use `futureNlnCheckNonCached` in `futureNlnCheckMem`

File: src/Fantomas.Tests/CommentTests.fs (hunk 1 above)
Comment (nojaf) [reply src/Fantomas.Tests/CommentTests.fs]: Reverted

File: src/Fantomas/Context.fs (hunk 2 above)
Comment (nojaf) [reply src/Fantomas/Context.fs]: Not used anymore

**Comment by nojaf** [comment]
@jindraivanek ready for review, let me know what you think.

**Pull request merged**